modules/platforms/cpp/thin-client/src/impl/transactions/transactions_impl.h

```diff
@@ -0,0 +1,268 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef TRANSACTIONS_IMPL_H
+#define TRANSACTIONS_IMPL_H
+
+#include "impl/data_router.h"
+#include "ignite/thin/transactions/transaction_consts.h"
+
+using namespace ignite::thin::transactions;
+using namespace ignite::common::concurrent;
+
+namespace ignite
+{
+    namespace impl
+    {
+        namespace thin
+        {
+            namespace transactions
+            {
+                /* Forward declaration. */
+                class TransactionsImpl;
+
+                /* Forward declaration. */
+                class TransactionImpl;
+
+                typedef SharedPointer<TransactionsImpl> SP_TransactionsImpl;
+                typedef SharedPointer<TransactionImpl> SP_TransactionImpl;
+
+                /**
+                 * Ignite transactions implementation.
+                 *
+                 * This is an entry point for Thin C++ Ignite transactions.
+                 */
+                class TransactionImpl
+                {
+                    typedef ThreadLocalInstance<SP_TransactionImpl> TL_TXID;
```

Comment:
The naming is bad, looks like a constant or macro. Also, typedef itself is not really needed - the type only used in 2 places.